File: android/app/src/androidTest/java/org/getlantern/lantern/test/BaseTest.java
package org.getlantern.lantern.test;

import android.content.Context;

import androidx.test.platform.app.InstrumentationRegistry;

import org.junit.After;
import org.junit.Before;

import java.io.File;
import java.util.Random;

public class BaseTest {
    protected File tempDir = null;

    protected Context getTargetContext() {
        return InstrumentationRegistry.getInstrumentation().getTargetContext();
    }

    @Before
    public void setupTempDir() {
        tempDir = new File(
                getTargetContext().getCacheDir(),
                Long.valueOf(new Random().nextLong()).toString()
        );
        tempDir.mkdirs();
    }

    @After
    public void deleteTempDir() {
        deleteDirectory(tempDir);
    }

    void deleteDirectory(File dir) {
        File[] allContents = dir.listFiles();
        if (allContents != null) {
            for (File file : allContents) {
                deleteDirectory(file);
            }
        }
        dir.delete();
    }
}

File: android/app/src/androidTest/kotlin/org/getlantern/lantern/test/BaseTest.kt
package org.getlantern.lantern.test

import android.content.Context
import androidx.test.platform.app.InstrumentationRegistry
import org.junit.After
import org.junit.Before
import java.io.File
import java.util.Random

open class BaseTest {

    var tempDir : File? = null

    fun getTargetContext(): Context {
        return InstrumentationRegistry.getInstrumentation().getTargetContext()
    }

    @Before
    fun setupTempDir() {
        tempDir = File(
                getTargetContext().getCacheDir(),
                Random().nextLong().toString()
        )
        tempDir?.mkdirs()
    }

    @After
    fun deleteTempDir() {
        deleteDirectory(tempDir!!)
    }

    fun deleteDirectory(dir: File) {
        var allContents = dir.listFiles()
        if (allContents != null) {
            for (file in allContents) {
                deleteDirectory(file)
            }
        }
        dir.delete()
    }
}
